src/formats/MMTF.cpp

```diff
@@ -0,0 +1,118 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+
+#include "chemfiles/formats/MMTF.hpp"
+
+#include "chemfiles/ErrorFmt.hpp"
+#include "chemfiles/File.hpp"
+#include "chemfiles/Frame.hpp"
+#include "chemfiles/utils.hpp"
+#include "chemfiles/warnings.hpp"
+
+using namespace chemfiles;
+
+template<> FormatInfo chemfiles::format_information<MMTFFormat>() {
+    return FormatInfo("MMTF").with_extension(".mmtf").description(
+        "MMTF (RCSB Protein Data Bank) binary format"
+    );
+}
+
+MMTFFormat::MMTFFormat(const std::string& path, File::Mode mode)
+  : file_(path, mode),
+    modelIndex_(0), chainIndex_(0), groupIndex_(0), atomIndex_(0),
+    atomSkip_(0) {
+}
+
+size_t MMTFFormat::nsteps() {
+    return file_->numModels;
+}
+
+void MMTFFormat::read_step(const size_t step, Frame& frame) {
+
+    if (step >= nsteps()) {
+        throw out_of_bounds("Step size of {} is larger than nsteps {}", step, nsteps());
+    }
+
+    modelIndex_ = 0;
+    chainIndex_ = 0;
+    groupIndex_ = 0;
+    atomIndex_  = 0;
+    atomSkip_   = 0;
+
+    // Fast-forward, keeping all indecies updated
+    while(modelIndex_ != step) {
```

Comment:
Does this work if I call `read_step(10)` and then `read_step(2)`? Or should the code reset the position in the file ?

Reply:
The MMTF-c parser reads the entire file at once, so there's no position to reset. If you have suggestions, I'm all ears.

Reply:
So the whole file is always in memory ? In that case I'm fine with this 😃

Reply:
From my understanding, yes. The files are quite small so the parser loads the whole file at once.